instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt

```diff
@@ -23,22 +25,161 @@ class ExplorationPlayerTest {
   }
 
   @Test
-  fun testExploration_prototypeExploration_toolbarTitle_isDisplayedSuccessfully() {
+  fun testPlayExploration_prototypeExploration_playedFullyThrough_finishesSuccessfully() {
     navigateToPrototypeExploration()
-    assertThat(device.findObjectByRes("exploration_toolbar_title")).isNotNull()
+
+    // Play through all interactions.
+    playContinueInteraction()
+    playFractionInputInteraction()
+    playMultipleChoiceIntearction1()
+    playMultipleChoiceIntearction2()
+    playItemSelectionInteraction()
+    playNumericInputInteraction()
+    playRatioInputInteraction()
+    playTextInputInteraction()
+    playDragAndDropInteraction()
+    playDragDropMergeInteraction()
+    playEndExplorationInteraction()
+
+    // Assert Topic Completed.
+    scrollRecyclerViewTextIntoView("Chapter 1: Prototype Exploration")
+    val chapterCompletedTick = device.findObjectByDesc(
+      "Chapter 1 with title Prototype Exploration is completed"
+    )!!
+    assertThat(chapterCompletedTick).isNotNull()
+  }
+
+  // TODO(#3697): Update e2e tests when backend support is introduced.
+  @Test
+  @Ignore("Need backend connection support to test the ImageRegionSelectionInteraction")
+  fun testPlayExploration_imageRegionInteractionExp_playedFullyThrough_finishesSuccessfully() {
+    navigateToImageRegionSelectionInteraction()
+
+    // Image Region Selection Interaction.
+    val imageSelectionView = device.findObjectByRes("interaction_container_frame_layout")
+    device.waitForRes("image_click_interaction_image_view")
+    imageSelectionView!!.children!!.get(2)!!.click()
+    device.findObjectByText("SUBMIT")!!.click()
```

Comment:
Here I'm using `!!.` instead of `?.`
So that in case the view doesn't exist the test fails immediately, else the test runs as normal till it reaches the assertion and then gives a failure which might be confusing

Reply:
I suggest returning only non-nullable types from your helper class and utilize checkNotNull with clearer error messages if an object isn't found. That'll give you the quick-failure you're looking for with clearer context, and doesn't even give the choice to test authors to potentially avoid the null-check (since we do want things to be really crisp).

Reply:
Done